RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1T.cc

```diff
@@ -0,0 +1,383 @@
+  theL1MinQuality(iConfig.getParameter<unsigned int>("L1MinQuality")),
+  useOfflineSeed(iConfig.getUntrackedParameter<bool>("UseOfflineSeed", false)),
+  useUnassociatedL1(iConfig.existsAs<bool>("UseUnassociatedL1") ? 
+		    iConfig.getParameter<bool>("UseUnassociatedL1") : true),
+  centralBxOnly_( iConfig.getParameter<bool>("CentralBxOnly") )
+  {
+
+  muCollToken_ = consumes<MuonBxCollection>(theSource);
+
+  if(useOfflineSeed) {
+    theOfflineSeedLabel = iConfig.getUntrackedParameter<InputTag>("OfflineSeedLabel");
+    offlineSeedToken_ = consumes<edm::View<TrajectorySeed> >(theOfflineSeedLabel);
+  }
+  
+  // service parameters
+  ParameterSet serviceParameters = iConfig.getParameter<ParameterSet>("ServiceParameters");
+  
+  // the services
+  theService = new MuonServiceProxy(serviceParameters);
+
+  // the estimator
+  theEstimator = new Chi2MeasurementEstimator(10000.);
+
+
+  produces<L2MuonTrajectorySeedCollection>(); 
+}
+
+// destructor
+L2MuonSeedGeneratorFromL1T::~L2MuonSeedGeneratorFromL1T(){
+  if (theService)   delete theService;
+  if (theEstimator) delete theEstimator;
+}
+
+void
+L2MuonSeedGeneratorFromL1T::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("GMTReadoutCollection", edm::InputTag("")); // to be removed
+  desc.add<edm::InputTag>("InputObjects",edm::InputTag("hltGmtStage2Digis"));
+  desc.add<string>("Propagator", "");
+  desc.add<double>("L1MinPt",-1.);
+  desc.add<double>("L1MaxEta",5.0);
+  desc.add<unsigned int>("L1MinQuality",0);   
+  desc.addUntracked<bool>("UseOfflineSeed",false);
+  desc.add<bool>("UseUnassociatedL1", true);
+  desc.add<bool>("CentralBxOnly", true);
+  desc.addUntracked<edm::InputTag>("OfflineSeedLabel", edm::InputTag(""));
+
+  edm::ParameterSetDescription psd0;
+  psd0.addUntracked<std::vector<std::string>>("Propagators", {
+	  "SteppingHelixPropagatorAny"
+  });
+  psd0.add<bool>("RPCLayers", true);
+  psd0.addUntracked<bool>("UseMuonNavigation", true);
+  desc.add<edm::ParameterSetDescription>("ServiceParameters", psd0);
+  descriptions.add("L2MuonSeedGeneratorFromL1T",desc);
+}
+
+void L2MuonSeedGeneratorFromL1T::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  const std::string metname = "Muon|RecoMuon|L2MuonSeedGeneratorFromL1T";
+  MuonPatternRecoDumper debug;
+
+  auto_ptr<L2MuonTrajectorySeedCollection> output(new L2MuonTrajectorySeedCollection());
+  
+  edm::Handle<MuonBxCollection> muColl;
+  iEvent.getByToken(muCollToken_, muColl);
+  LogTrace(metname) << "Number of muons " << muColl->size() << endl;
+
+  edm::Handle<edm::View<TrajectorySeed> > offlineSeedHandle;
+  vector<int> offlineSeedMap;
+  if(useOfflineSeed) {
+    iEvent.getByToken(offlineSeedToken_, offlineSeedHandle);
+    LogTrace(metname) << "Number of offline seeds " << offlineSeedHandle->size() << endl;
+    offlineSeedMap = vector<int>(offlineSeedHandle->size(), 0);
+  }
+
+  if (muColl.isValid()){
```

Comment:
The collection not present is more indicative of incorrect configuration.
Why do we need to catch this exception here?
What's a good reason for the MuonBxCollection producer not to make an empty collection?